http-client-spi/src/main/java/software/amazon/awssdk/http/DefaultSdkHttpFullRequest.java

```diff
@@ -172,12 +169,35 @@ public String toString() {
         private String host;
         private Integer port;
         private String path;
-        private Map<String, List<String>> queryParameters = new LinkedHashMap<>();
+
+        private boolean queryParametersAreFromToBuilder;
+        private Map<String, List<String>> queryParameters;
+
         private SdkHttpMethod httpMethod;
-        private Map<String, List<String>> headers = new LinkedHashMap<>();
+
+        private boolean headersAreFromToBuilder;
+        private Map<String, List<String>> headers;
+
         private ContentStreamProvider contentStreamProvider;
 
         Builder() {
+            queryParameters = new LinkedHashMap<>();
+            queryParametersAreFromToBuilder = false;
```

Comment:
Substance over beauty!

Reply:
Hah, I wrote a class to do this originally, but it just made things harder to read. If we find ourselves doing this a lot we can put it somewhere else...